compiler/damlc/daml-prim-src/GHC/Classes.daml

```diff
@@ -72,45 +72,44 @@ default ()              -- Double isn't available yet
 class IP (x : Symbol) a | x -> a where
   ip : a
 
-{- $matching_overloaded_methods_in_rules
-
-Matching on class methods (e.g. `(==)`) in rewrite rules tends to be a bit
-fragile. For instance, consider this motivating example from the `bytestring`
-library,
-
-> break :: (Word8 -> Bool) -> ByteString -> (ByteString, ByteString)
-> breakByte :: Word8 -> ByteString -> (ByteString, ByteString)
-> {-# RULES "break -> breakByte" forall a. break (== x) = breakByte x #-}
-
-Here we have two functions, with `breakByte` providing an optimized
-implementation of `break` where the predicate is merely testing for equality
-with a known `Word8`. As written, however, this rule will be quite fragile as
-the `(==)` class operation rule may rewrite the predicate before our `break`
-rule has a chance to fire.
-
-For this reason, most of the primitive types in `base` have 'Eq' and 'Ord'
-instances defined in terms of helper functions with inlinings delayed to phase
-1. For instance, `Word8`\'s `Eq` instance looks like,
-
-> instance Eq Word8 where
->     (==) = eqWord8
->     (/=) = neWord8
->
-> eqWord8, neWord8 :: Word8 -> Word8 -> Bool
-> eqWord8 (W8# x) (W8# y) = ...
-> neWord8 (W8# x) (W8# y) = ...
-> {-# INLINE [1] eqWord8 #-}
-> {-# INLINE [1] neWord8 #-}
-
-This allows us to save our `break` rule above by rewriting it to instead match
-against `eqWord8`,
-
-> {-# RULES "break -> breakByte" forall a. break (`eqWord8` x) = breakByte x #-}
-
-Currently this is only done for '(==)', '(/=)', '(<)', '(<=)', '(>)', and '(>=)'
-for the types in "GHC.Word" and "GHC.Int".
--}
-
+-- $matching_overloaded_methods_in_rules
```

Comment:
This works around a parse error with the flag enabled.

Reply:
This seems slightly worrying. What exactly causes the parse errors here? I would hope that blog comments aren’t broken in general. It would also be good to add a comment about this.

Reply:
I *think* it relates to the embedded comment. The parse error emerges on the next '`,`'. I hope it is a rare bug - if it breaks more generally we'll have no choice but to work it out and fix it or turn it off.